buildSrc/src/main/kotlin/Java9Modularity.kt

```diff
@@ -0,0 +1,116 @@
+            val targetDir = compileKotlinTask.destinationDirectory.dir("../java9")
+
+            // Use a Java 11 compiler for the module-info.
+            javaCompiler.set(javaToolchains.compilerFor {
+                languageVersion.set(JavaLanguageVersion.of(11))
+            })
+
+            // Always compile kotlin classes before the module descriptor.
+            dependsOn(compileKotlinTask)
+
+            // Add the module-info source file.
+            // Note that we use the parent dir and an include filter,
+            // this is needed for Gradle's module detection to work in
+            // org.gradle.api.tasks.compile.JavaCompile.createSpec
+            source(sourceFile.parentFile)
+            include { it.file == sourceFile }
+
+            // The Kotlin compiler will parse and check module dependencies,
+            // but it currently won't compile to a module-info.class file.
+            // Note that module checking only works on JDK 9+,
+            // because the JDK built-in base modules are not available in earlier versions.
+            val javaVersion = compileKotlinTask.kotlinJavaToolchain.javaVersion.orNull
+            if (javaVersion?.isJava9Compatible == true) {
+                logger.info("Module-info checking is enabled; $compileKotlinTask is compiled using Java $javaVersion")
+            } else {
+                logger.info("Module-info checking is disabled")
+                // Exclude the module-info.java source file from the Kotlin compile task,
+                // to prevent compile errors when resolving the module graph.
+                compileKotlinTask.exclude { it.file == sourceFile }
+            }
+
+            // Set the task outputs and destination directory
+            outputs.dir(targetDir)
+            destinationDirectory.set(targetDir)
+
+            // Configure JVM compatibility
+            sourceCompatibility = JavaVersion.VERSION_1_9.toString()
+            targetCompatibility = JavaVersion.VERSION_1_9.toString()
+
+            // Set the Java release version.
+            options.release.set(9)
+
+            // Ignore warnings about using 'requires transitive' on automatic modules.
+            // not needed when compiling with recent JDKs, e.g. 17
+            options.compilerArgs.add("-Xlint:-requires-transitive-automatic")
+
+            // Patch the compileKotlinJvm output classes into the compilation so exporting packages works correctly.
+            val kotlinCompileDestinationDir = compileKotlinTask.destinationDirectory.asFile.get()
+            options.compilerArgs.addAll(listOf("--patch-module", "$moduleName=$kotlinCompileDestinationDir"))
+
+            // Use the classpath of the compileKotlinJvm task.
+            // Also ensure that the module path is used instead of classpath.
+            classpath = compileKotlinTask.libraries
+            modularity.inferModulePath.set(true)
+        }
+
+        tasks.getByName<Jar>(target.artifactsTaskName) {
```

Comment:
Please change it to
```suggestion
        tasks.named<Jar>(target.artifactsTaskName) {
```
You're triggering task unconditional task configuration even on running `./gradlew help`:
<img width="703" alt="Screenshot 2023-02-16 at 14 25 16" src="https://user-images.githubusercontent.com/1221596/219377063-ad11e632-137a-4556-b4ce-4c188c335a3b.png">